src/webots/vrml/WbProtoManager.hpp

```diff
@@ -135,20 +141,24 @@ class WbProtoManager : public QObject {
   // searches for proto model according to:
   // 1. First in the session list (i.e., PROTO discovered by navigating the PROTO tree stemming from a world file
   // 2. If the first fails, it searches among the known official Webots PROTO (proto-list.xml)
-  WbProtoModel *findModel(const QString &modelName, const QString &worldPath, QStringList baseTypeList = QStringList());
+  WbProtoModel *findModel(const QString &modelName, const QString &worldPath, const QString &parentFilePath,
+                          QStringList baseTypeList = QStringList());
```

Comment:
we should avoid passing objects by value
```suggestion
                          const QStringList &baseTypeList = QStringList());
```

Reply:
I don't believe we can since the value is changed and propagated further down, unless we remove the default value from the declaration

Reply:
It can't be const, and to pass by reference we do indeed need to remove the default value, which adds a whole lot of hanging variables, I'm 50/50 on whether passing by reference really brings us much in terms for performance/cleanliness in this case: https://github.com/cyberbotics/webots/pull/4860/commits/9d58bca3ac6aa85ea56878fd31e3ce709d228991
What do you think?

Reply:
The default value can be set even if you pass the argument by reference.
What doesn't work?

Reply:
I'm getting: 

```
cannot bind non-const lvalue reference of type ‘QStringList&’ {aka ‘QList<QString>&’} to an rvalue of type ‘QStringList’ {aka ‘QList<QString>’}
  145 |                           QStringList &baseTypeList = QStringList());
```

Reply:
You should also add the `const

Reply:
Ah nevermind, I've read wrong, I thought you suggested WbProtoModel should receive it by reference, but that one can't because it appends items to the list. `readmodel` and `findmodel` indeed can